pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel_test.go

```diff
@@ -408,6 +495,7 @@ func TestReconciler_ReconcileKind(t *testing.T) {
 		},
 	}
 	for n, tc := range testCases {
+		ctx, fakeEventingClient := fakeeventingclient.With(context.Background(), tc.imc)
```

Comment:
Won't this already be injected in all tests, due to the "magic" `fakeeventingclient "knative.dev/eventing/pkg/client/injection/client/fake"` import?

Reply:
Ahhh no, ReconcileKind is called manually in those tests.

/lgtm

Reply:
Yeah, I broke it into two tests, because what I really wanted to do was to see how the resources are handled (using the magic factory) and how the channels are configured by manually calling it. Our test infra could make this easier :)